definitions/procedures/installer/upgrade.rb

```diff
@@ -6,7 +6,11 @@ class Upgrade < ForemanMaintain::Procedure
 
     def run
       assumeyes_val = @assumeyes.nil? ? assumeyes? : @assumeyes
-      feature(:installer).run(@arguments, :interactive => !assumeyes_val)
+      if package_version('foreman-installer').version >= '2.1'
+        feature(:installer).run(@arguments, :interactive => !assumeyes_val)
```

Comment:
IIUC, in this case installer command will run without `--disable-system-checks` option. 
@patilsuraj767, @upadhyeammit,  will it be okay to run it without this flag in latest foreman (>=2.1)?

Reply:
Confirmed that this flag is having `false` as default value. 

@patilsuraj767 as discussed let's keep `upgrade` method only with `:interactive => !assumeyes_val` change so that `installer` feature will handle `--upgrade` flag based on condition and also consider `--disable-system-checks` option.

Reply:
Done